cryptoki/src/mechanism/mod.rs

```diff
@@ -1073,6 +1079,13 @@ impl From<&Mechanism<'_>> for CK_MECHANISM {
             | Mechanism::Sha512RsaPkcsPss(params) => make_mechanism(mechanism, params),
             Mechanism::RsaPkcsOaep(params) => make_mechanism(mechanism, params),
             Mechanism::Ecdh1Derive(params) => make_mechanism(mechanism, params),
+            Mechanism::Eddsa(params) => match params {
+                Some(params) => make_mechanism(mechanism, params),
+                None => CK_MECHANISM {
+                    mechanism,
+                    ..Default::default()
```

Comment:
Continued:

This line made me suspicious. Maybe it's just me but I think this would be more readable (basically the same code as for `Mechanism::HkdfKeyGen`):

```suggestion
                pParameter: null_mut(),
                ulParameterLen: 0,
```

I wonder if this is just me, though, and yeah, I'm aware that this makes the code longer :sweat_smile: